### ISSUE TITLE
geoInterruptedHomolosine is missing an invert() method

### ISSUE THREAD

**Issue by apezel** [d3/d3-geo-projection#97]
I don't know if it's normal but ```d3.geoInterruptedHomolosine()``` misses an ```.invert()``` method.

**Comment by Fil** [comment]
Not all projections have an invert — see https://github.com/d3/d3-geo-projection/issues/85 for a general solution.
Of course if we can invert the equations it's better, but it's not always possible.

**Comment by apezel** [comment]
Ok I understand. But in this case, if i use ```d3.geoHomolosine()``` with manually defined lobes i can use the ```invert()``` method and it seems to work correctly. Do you mean that this method returns wrong coordinates ?

**Comment by apezel** [comment]
Oups sorry. Forget my comment. Defining lobes on d3.geoHomolosine doesn't work at all.

**Comment by apezel** [comment]
@Fil Actually, i think it may be a bug : 

If i defined the projection projection as : 
```
d3.geoInterrupt(d3.geoHomolosineRaw, [[ [[-180, 0], [-100, 90], [ -40, 0]], [[ -40, 0], [ 30, 90], [ 180, 0]] ], [ [[-180, 0], [-160, -90], [-100, 0]], [[-100, 0], [ -60, -90], [ -20, 0]], [[ -20, 0], [ 20, -90], [ 80, 0]], [[ 80, 0], [ 140, -90], [ 180, 0]] ]])
```
there's no ```invert()``` method neither. When i look into interrupted/index.js, it seems that the projection is initialized before that the ```invert()``` method get attached to ```forward``` :  
```
var p = projection(forward),
      stream_ = p.stream;

  function forward(lambda, phi) {
    var sign = phi < 0 ? -1 : +1, lobe = lobes[+(phi < 0)];
    for (var i = 0, n = lobe.length - 1; i < n && lambda > lobe[i][2][0]; ++i);
    var p = project(lambda - lobe[i][1][0], phi);
    p[0] += project(lobe[i][1][0], sign * phi > sign * lobe[i][0][1] ? lobe[i][0][1] : phi)[0];
    return p;
  }

  // Assumes mutually exclusive bounding boxes for lobes.
  if (project.invert) forward.invert = function(x, y) {
    var bound = bounds[+(y < 0)], lobe = lobes[+(y < 0)];
    for (var i = 0, n = bound.length; i < n; ++i) {
      var b = bound[i];
      if (b[0][0] <= x && x < b[1][0] && b[0][1] <= y && y < b[1][1]) {
        var p = project.invert(x - project(lobe[i][1][0], 0)[0], y);
        p[0] += lobe[i][1][0];
        return pointEqual(forward(p[0], p[1]), [x, y]) ? p : null;
      }
    }
};

```

If i move the initialization to the end of the block (like this), i can get the projection.invert method to work perfectly  : 

```
  function forward(lambda, phi) {
    var sign = phi < 0 ? -1 : +1, lobe = lobes[+(phi < 0)];
    for (var i = 0, n = lobe.length - 1; i < n && lambda > lobe[i][2][0]; ++i);
    var p = project(lambda - lobe[i][1][0], phi);
    p[0] += project(lobe[i][1][0], sign * phi > sign * lobe[i][0][1] ? lobe[i][0][1] : phi)[0];
    return p;
  }

  // Assumes mutually exclusive bounding boxes for lobes.
  if (project.invert) forward.invert = function(x, y) {
    var bound = bounds[+(y < 0)], lobe = lobes[+(y < 0)];
    for (var i = 0, n = bound.length; i < n; ++i) {
      var b = bound[i];
      if (b[0][0] <= x && x < b[1][0] && b[0][1] <= y && y < b[1][1]) {
        var p = project.invert(x - project(lobe[i][1][0], 0)[0], y);
        p[0] += lobe[i][1][0];
        return pointEqual(forward(p[0], p[1]), [x, y]) ? p : null;
      }
    }
};

var p = projection(forward),
      stream_ = p.stream;
```

**Comment by Fil** [comment]
Yes I just fixed it the same way you did in http://blockbuilder.org/Fil/cb3718040261307ef04b4031615e0c5c

Would you like to submit a PR?

**Comment by mbostock** [comment]
Fixed by #98.
